go-controller/pkg/cni/cni.go

```diff
@@ -174,9 +174,26 @@ func (pr *PodRequest) cmdAddWithGetCNIResultFunc(kubeAuth *KubeAPIAuth, clientse
 			if err != nil {
 				return nil, err
 			}
-			if _, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo); err != nil {
+			primaryUDNResult, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo)
+			if err != nil {
 				return nil, err
 			}
+
+			originalNumberOfIfaces := len(response.Result.Interfaces)
+			response.Result.Interfaces = append(response.Result.Interfaces, primaryUDNResult.Interfaces...)
+			for i := range primaryUDNResult.IPs {
+				ifaceIPConfig := primaryUDNResult.IPs[i].Copy()
+
+				if ifaceIPConfig.Interface != nil {
+					newIDX := *ifaceIPConfig.Interface + originalNumberOfIfaces
+					ifaceIPConfig.Interface = &newIDX
```

Comment:
heads up: you can use 
```
ifaceIPConfig.Interface = current.Int(*ifaceIPConfig.Interface + originalNumberOfIfaces)
```

Reply:
I don't follow this code; won't we find a nil pointer if `ifaceIPConfig.Interface` is nil ?

That's what I'm trying to avoid. I am probably missing something.